python/cudf/cudf/core/column/string.py

```diff
@@ -5009,9 +5010,22 @@ def find_and_replace(
         """
         Return col with *to_replace* replaced with *value*
         """
-        to_replace = column.as_column(to_replace, dtype=self.dtype)
-        replacement = column.as_column(replacement, dtype=self.dtype)
-        return libcudf.replace.replace(self, to_replace, replacement)
+
+        to_replace_col = column.as_column(to_replace)
+        if to_replace_col.null_count == len(to_replace_col):
+            to_replace_col = to_replace_col.astype(self.dtype)
+
+        replacement_col = column.as_column(replacement)
+        if replacement_col.null_count == len(replacement_col):
+            replacement_col = replacement_col.astype(self.dtype)
```

Comment:
This is essentially handling if either of `to_replace` or `replacement` are all `None`, then the column types being constructed would be `float64` and thus to prevent the type-mismatch and early stopping which results in type-mismatch at line 5022-5026

Reply:
Got it, so really this is handling the case where `to_replace` or `replacement` are a column of length 1 and are all `null`, something like: `[None]`?

Reply:
Yep, exactly.

Reply:
Could you add a comment along those lines?

Reply:
Done